src/System.Management.Automation/engine/ShellVariable.cs

```diff
@@ -668,7 +668,7 @@ private void SetValue(object value)
         {
             // Check to see if the variable is writable
 
-            if ((_options & (ScopedItemOptions.ReadOnly | ScopedItemOptions.Constant)) != ScopedItemOptions.None)
+            if (IsReadOnly || IsConstant)
```

Comment:
Your change is slightly more readable, but this code could be on a hot path (setting variables) so I don't see a good reason to change it.